### PR TITLE
fix: Manual Override only on out-of-range CT; remove Sync Grace Period

### PR DESCRIPTION
**Pull request by mholubinka1**
## Summary
- Redefine Manual Override detection: a light is only considered manually overridden when its CT falls outside the configured operating range [`Coolest`, `Warmest`], or when in XY mode and the XY→CT conversion fails or yields an out-of-range value. Previously, any divergence from Expected Light triggered Manual Override — causing IKEA TRADFRI GU10 bulbs (which silently revert to their power-on CT ~13s after a sync) to permanently escape app control.
- Add drift re-sync: when a light's network CT is in-range but differs from Expected Light by more than 10 mired, `SyncRequired` is set, triggering an immediate re-sync on the next cycle.
- Remove Sync Grace Period and `LightPowerState.Syncing`: the former retry/confirmation state machine is replaced by self-healing drift correction. Sync is now fire-and-forget over `BasicTransitionDuration`.
- Suppress drift and Manual checks during `LightPowerState.Transitioning`.

## Test plan
- [ ] 13 new BDD tests in `LightControlPairManualOverrideTests` cover: TRADFRI revert scenario, CT boundary cases (above/below/at Warmest and Coolest), drift tolerance (>10 mired triggers sync, ≤10 does not), XY mode (in-range → HueShift + drift sync, failed conversion → Manual, out-of-range → Manual), transitioning suppression, Off→On turn-on sync, and Reset behaviour
- [ ] `LightControllerRefreshTests` updated: removed `SyncGracePeriod` from options, corrected CT fixture from 500→400 (500 is now correctly out-of-range and Manual)
- [ ] `LightControlPairSyncGraceTests` deleted (behaviour removed)
- [ ] Deploy to Pi and confirm TRADFRI GU10 kitchen lights no longer trigger false Manual Override after power-on

🤖 Generated with Claude Code

### CODE REVIEW

File: HueShift2/HueShift2/Control/LightSynchroniser.cs [hunk 1]
```diff
@@ -31,38 +31,28 @@ public async Task Synchronise(IReadOnlyDictionary<string, LightControlPair> ligh
...
             }
 
-            if (syncCommands.Any()) await Dispatch(syncCommands, lights, duration, currentTime, logSync: true);
-            if (retrySyncCommands.Any()) await Dispatch(retrySyncCommands, lights, duration, currentTime, logSync: false);
+            if (syncCommands.Any()) await Dispatch(syncCommands, lights, duration);
         }
 
-        private async Task Dispatch(IDictionary<string, LightCommand> commands, IReadOnlyDictionary<string, LightControlPair> lights, TimeSpan duration, DateTime currentTime, bool logSync)
+        private async Task Dispatch(IDictionary<string, LightCommand> commands, IReadOnlyDictionary<string, LightControlPair> lights, TimeSpan duration)
         {
```
[the remark this answers is not in the document]
Comment (mholubinka1) [reply HueShift2/HueShift2/Control/LightSynchroniser.cs]: Fixed. Dispatch now calls ExecuteCommand instead of ExecuteInstantaneousCommand, which sets PowerState to Transitioning for BasicTransitionDuration. Since Synchronise already skips Transitioning lights, the in-flight sync window is automatically respected and re-sync spam is prevented. LogSync is emitted after all ExecuteCommand calls so ExpectedLight reflects the synced target; in practice all lights in a sync batch follow the same time-based CT schedule and share the same target.

File: HueShift2/HueShift2/Control/LightSynchroniser.cs [hunk 2]
```diff
@@ -31,38 +31,28 @@ public async Task Synchronise(IReadOnlyDictionary<string, LightControlPair> ligh
...
+        private async Task Dispatch(IDictionary<string, LightCommand> commands, IReadOnlyDictionary<string, LightControlPair> lights, TimeSpan duration, DateTime currentTime)
         {
-            var lightNames = commands.Keys.Select(id => lights[id].Properties.Name);
             foreach (var (id, command) in commands)
             {
                 command.TransitionTime = duration;
-                lights[id].ExecuteInstantaneousCommand(command);
-                lights[id].ExecuteSync(duration, currentTime);
+                lights[id].ExecuteCommand(command, currentTime, TransitionType.Adaptive);
                 await client.SendCommandAsync(command, new[] { id });
```
[the remark this answers is not in the document]
Comment (mholubinka1) [reply HueShift2/HueShift2/Control/LightSynchroniser.cs:52]: Fixed. TransitionType.Sync already exists in the enum — changed from Adaptive to Sync.

File: HueShift2/HueShift2/Control/LightSynchroniser.cs [hunk 3]
```diff
@@ -31,38 +31,28 @@ public async Task Synchronise(IReadOnlyDictionary<string, LightControlPair> ligh
...
             }
-            if (logSync)
-            {
-                var targetKeys = commands.Keys.Select(id => lights[id].ExpectedLight.ToString()).Distinct().ToList();
-                var commonTarget = targetKeys.Count == 1 ? lights[commands.Keys.First()].ExpectedLight : null;
-                logger.LogSync(lightNames, commonTarget);
-            }
+            var lightNames = commands.Keys.Select(id => lights[id].Properties.Name);
+            logger.LogSync(lightNames, lights[commands.Keys.First()].ExpectedLight);
```
[the remark this answers is not in the document]
Comment (mholubinka1) [reply HueShift2/HueShift2/Control/LightSynchroniser.cs]: Fixed. Added a common-target check: if all commands in the batch share the same target CT, LogSync emits that target; otherwise it passes null. In practice all sync-batch lights follow the same time-based schedule and share a target, but this is now correct in the general case.

File: HueShift2/HueShift2/Control/LightControlPair.cs [hunk 4]
```diff
@@ -58,56 +54,32 @@ public void RefreshTransition(bool isOn, DateTime currentTime)
...
-            this.SyncFailed = false;
             var isOn = networkLight.DeterminePowerState() == LightPowerState.On;
             var wasOff = this.PowerState == LightPowerState.Off;
             if (isOn)
             {
                 switch (this.AppControlState)
                 {
                     case LightControlState.HueShift:
-                        if (this.PowerState == LightPowerState.Syncing)
+                        if (this.PowerState == LightPowerState.On)
```
[the remark this answers is not in the document]
Comment (mholubinka1) [reply HueShift2/HueShift2/Control/LightControlPair.cs:73]: Ignored. The PowerState == On gate before RefreshTransition is intentional. On the first poll after turn-on, PowerState is still Off, so the drift/manual checks are skipped and the isOn && wasOff path sets SyncRequired instead. A light that just powered on is at its factory default, not a user-set value, so syncing it back to the expected CT is the correct response — flagging it as Manual Override on turn-on would be wrong.

File: HueShift2/HueShift2/Control/LightSynchroniser.cs [hunk 5]
```diff
@@ -31,38 +31,30 @@ public async Task Synchronise(IReadOnlyDictionary<string, LightControlPair> ligh
...
             }
-            if (logSync)
-            {
-                var targetKeys = commands.Keys.Select(id => lights[id].ExpectedLight.ToString()).Distinct().ToList();
-                var commonTarget = targetKeys.Count == 1 ? lights[commands.Keys.First()].ExpectedLight : null;
-                logger.LogSync(lightNames, commonTarget);
-            }
+            var lightNames = commands.Keys.Select(id => lights[id].Properties.Name);
+            var distinctTargets = commands.Values.Select(c => c.ColorTemperature).Distinct().ToList();
+            var logTarget = distinctTargets.Count == 1 ? lights[commands.Keys.First()].ExpectedLight : null;
```
[the remark this answers is not in the document]
Comment (mholubinka1) [reply HueShift2/HueShift2/Control/LightSynchroniser.cs]: Fixed. Switched from ColorTemperature-only key to ExpectedLight.ToString() composite key, which covers CT, XY, brightness, and all other colour state — same pattern the previous implementation used.

File: docs/adr/0001-manual-override-detection-and-sync-model.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+Override by comparing Network Light to Expected Light: any divergence while `PowerState == On`
+and `AppControlState == HueShift` was treated as a user override. This caused TRADFRI bulbs to
+be permanently locked out of app control on every power-on cycle, requiring a physical toggle
+to recover, even though the user had not touched them.
+
+We redefined Manual Override as: a light whose CT mode value falls outside the configured
+operating range [Coolest, Warmest] (strict, no tolerance), or whose XY mode state either
+cannot be converted to a CT value or converts to one outside that range. CT values within the
+range — including a TRADFRI reverting to its 454 mired default — are treated as Drift, not
+Manual Override.
```
[the remark this answers is not in the document]
Comment (mholubinka1) [reply docs/adr/0001-manual-override-detection-and-sync-model.md]: Fixed. Added a sentence to the ADR documenting that a light reporting neither CT nor XY (e.g. HS or an unknown colour mode) is also treated as Manual Override, since there is no CT value to reason about.

File: HueShift2/HueShift2/Control/LightControlPair.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (mholubinka1) [reply HueShift2/HueShift2/Control/LightControlPair.cs:73]: Ignored. The one-poll delay after a transition expires is inherent to the single-pass state machine — RefreshTransition updates PowerState after the drift/manual checks in the same Refresh call, so the expired transition is cleared at the end of that call and checks resume on the very next poll. With BasicTransitionDuration = 2s and a ~5s poll interval, the extra suppression window is at most one poll cycle, which is acceptable.

File: HueShift2/HueShift2/Control/LightSynchroniser.cs [hunk 7]
```diff
@@ -31,38 +31,30 @@ public async Task Synchronise(IReadOnlyDictionary<string, LightControlPair> ligh
...
+        private async Task Dispatch(IDictionary<string, LightCommand> commands, IReadOnlyDictionary<string, LightControlPair> lights, TimeSpan duration, DateTime currentTime)
         {
-            var lightNames = commands.Keys.Select(id => lights[id].Properties.Name);
             foreach (var (id, command) in commands)
             {
                 command.TransitionTime = duration;
-                lights[id].ExecuteInstantaneousCommand(command);
-                lights[id].ExecuteSync(duration, currentTime);
+                lights[id].ExecuteCommand(command, currentTime, TransitionType.Sync);
                 await client.SendCommandAsync(command, new[] { id });
```
[the remark this answers is not in the document]
Comment (mholubinka1) [reply HueShift2/HueShift2/Control/LightSynchroniser.cs:52]: Ignored. If SendCommandAsync throws, the self-healing drift loop covers recovery: ExecuteCommand has updated ExpectedLight to the sync target, and NetworkLight will still report the old CT on the next poll, so HasDrifted returns true and re-flags SyncRequired automatically. Per-light error handling would duplicate what drift correction already provides.

File: HueShift2/HueShift2/Helpers/ExtensionMethods.cs [hunk 8]
```diff
@@ -102,6 +102,38 @@ public static LightCommand ToCommand(this AppLightState expectedLight)
...
+            var high = Math.Max(minCt, maxCt);
+            if (@this.ColorTemperature != null)
+                return @this.ColorTemperature.Value < low || @this.ColorTemperature.Value > high;
+            if (@this.ColorCoordinates != null)
+            {
+                if (!TryXyToCt(@this.ColorCoordinates, out var convertedCt)) return true;
+                return convertedCt < low || convertedCt > high;
+            }
+            return true;
+        }
```
[the remark this answers is not in the document]
Comment (mholubinka1) [reply HueShift2/HueShift2/Helpers/ExtensionMethods.cs:117]: Fixed. Both IsManualOverride and HasDrifted now switch on ColorMode.ToColourMode() as the primary discriminant, reading only the field that matches the active mode. A non-null but stale CT field on an XY-mode light is no longer consulted.

File: HueShift2/HueShift2/Helpers/ExtensionMethods.cs [hunk 9]
```diff
@@ -102,6 +102,38 @@ public static LightCommand ToCommand(this AppLightState expectedLight)
...
+            var expectedCt = expectedLight.Colour.ColourTemperature.Value;
+            if (@this.ColorTemperature != null)
+                return Math.Abs(@this.ColorTemperature.Value - expectedCt) > 10;
+            if (@this.ColorCoordinates != null)
+            {
+                if (!TryXyToCt(@this.ColorCoordinates, out var convertedCt)) return false;
+                return Math.Abs(convertedCt - expectedCt) > 10;
+            }
+            return false;
+        }
```
[the remark this answers is not in the document]
Comment (mholubinka1) [reply HueShift2/HueShift2/Helpers/ExtensionMethods.cs:129]: Fixed alongside IsManualOverride — HasDrifted now also dispatches on ColorMode.ToColourMode().

File: .gitattributes [hunk 10]
```diff
@@ -0,0 +1,9 @@
+# Normalize all text files to LF in the repo, CRLF in the working tree
+* text=auto eol=crlf
+
+# Explicitly mark binary files to prevent line-ending conversion
+*.png binary
+*.jpg binary
+*.ico binary
+*.dll binary
+*.exe binary
```
[the remark this answers is not in the document]
Comment (mholubinka1) [reply .gitattributes:13]: Fixed. Added explicit eol=lf overrides for .githooks/* and *.sh so shell scripts keep LF endings on checkout regardless of the global CRLF default.

File: docs/adr/0001-manual-override-detection-and-sync-model.md [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+reverting to its 454 mired default — are treated as Drift, not Manual Override.
+
+We also removed the Sync Grace Period and the `Syncing` power state. The previous model held
+lights in `Syncing` between command dispatch and Bridge confirmation, retried periodically, and
+promoted to Manual if the grace period expired without confirmation. This was the mechanism
+that made TRADFRI reverts fatal (the revert could race the confirmation window). Under the new
+model, Sync is fire-and-forget over `BasicTransitionDuration`. Drift (divergence from Expected
+Light by more than 10 mired while in-range and `PowerState == On`) is detected on every poll
+and triggers an immediate re-Sync — making correction self-healing without any grace period or
+explicit confirmation step.
```
[the remark this answers is not in the document]
Comment (mholubinka1) [reply docs/adr/0001-manual-override-detection-and-sync-model.md:24]: Fixed. Added a paragraph to the ADR documenting the Transition Settling Period — that every commanded transition sets PowerState to Transitioning for TransitionTime + 10s internally to suppress drift/manual checks while the Bridge is still settling, without changing the TransitionTime sent to the Bridge.

File: HueShift2/HueShift2/Control/LightSynchroniser.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (mholubinka1) [reply HueShift2/HueShift2/Control/LightSynchroniser.cs:52]: Ignored. The recovery path remains self-healing — once the Transitioning window (commanded + 10s) expires, the next Refresh detects drift and re-queues a sync. The settling period extends the maximum recovery delay on a send failure from ~2s to ~12s, which is acceptable for a background light-control service. Explicit per-command error handling would duplicate what drift correction already provides.

File: HueShift2/HueShift2.Tests/Control/LightControlPairManualOverrideTests.cs [hunk 12]
```diff
@@ -0,0 +1,223 @@
...
+        public void LightRevertsToInRangeCt_StaysHueShift()
+        {
+            // Given: a HueShift light synced to CT 309
+            var pair = LightOnAt(309);
+
+            // When: bridge reports CT 454 (TRADFRI power-on default — within [250, 454])
+            pair.Refresh(BridgeCt(454), Now, Coolest, Warmest);
+
+            // Then: light stays HueShift-controlled
+            Assert.Equal(LightControlState.HueShift, pair.AppControlState);
```
[the remark this answers is not in the document]
Comment (mholubinka1) [reply HueShift2/HueShift2.Tests/Control/LightControlPairManualOverrideTests.cs:53]: Fixed. Added Assert.True(pair.SyncRequired) to the TRADFRI tracer test — both halves of the in-range revert behavior are now asserted: AppControlState stays HueShift and SyncRequired is set for an immediate re-sync.
